docs/guide/n3iwue-installation.md

```diff
@@ -0,0 +1,147 @@
+
+# Installing N3IWUE 
+
+In this demo we will practice:
+
+- Installing N3IWUE
+- Configuring free5GC and N3IWUE
+- Running N3IWUE to connect free5GC
+
+## 1. Install N3IWUE VM
+
+Repeat the steps of cloning `free5gc` VM from the base VM, create a new VM for the N3IWUE:
+
+- You can refer to [Installing free5gc](https://free5gc.org/guide/3-install-free5gc/) to install free5gc VM.
+- Name the VM `n3iwue`, and create new MAC addresses for all network cards.
+- Make sure the VM has internet access and can log in using SSH.
+- Change the hostname to `n3iwue`.
+- Make the Host-only network interface have static IP address `192.168.56.103`.
+- Reboot the n3iwue VM, as well as the free5gc VM.
+- You can ping `192.168.56.101` from the n3iwue VM, and also `ping 192.168.56.103` from the free5gc VM.
+
+## 2. Install N3IWUE
+
+Search “free5gc/n3iwue” on the web, and get the [web site](https://github.com/free5gc/n3iwue).
+
+To download N3IWUE:
+```
+cd ~
+git clone https://github.com/free5gc/n3iwue.git
+cd ..
+```
+
+Update and upgrade n3iwue VM first:
+```
+sudo apt update
+sudo apt upgrade
+```
+
+Install required tools:
+```
+sudo apt install make
+sudo apt install libsctp-dev lksctp-tools
+sudo apt install iproute2
+```
+
+Install Golang (use latest version):
+```
+wget https://dl.google.com/go/go1.21.6.linux-amd64.tar.gz
+sudo tar -C /usr/local -zxvf go1.21.6.linux-amd64.tar.gz
+mkdir -p ~/go/{bin,pkg,src}
+echo 'export GOPATH=$HOME/go' >> ~/.bashrc
+echo 'export GOROOT=/usr/local/go' >> ~/.bashrc
+echo 'export PATH=$PATH:$GOPATH/bin:$GOROOT/bin' >> ~/.bashrc 
+echo 'export GO111MODULE=auto' >> ~/.bashrc
+source ~/.bashrc
+
+# check your go installation
+go version
+```
+
+Build N3iWUE
+```
+cd ~/n3iwue
+make
+```
+
+## 3. Use WebConsole to Add UE
+
+Open your web browser from your host machine, and enter the URL `http://192.168.56.101:5000`
+
+- On the login page, enter username `admin` and password `free5gc`.
+- Once logged in, widen the page until you see “Subscribers” on the left-hand side column.
+- Click on the `Subscribers` tab and then on the `New Subscriber` button
+    - Scroll down to `Operator Code Type` and change it from "OPc" to "OP".
+    - Synchronize these config between `n3iwue/config/n3ue.yaml` and `Subscriber`
+        - PLMNID (ex. 208930000001234)
+        - K
+        - SQN
+        - OP value (choose OP instead of OPC)
+    - Scroll all the way down and click on `Submit`.
+
+## 4. Setting N3IWF Config
+
+In free5gc VM, we need to edit one file:
+
+- `~/free5gc/config/n3iwfcfg.yaml`
+
+Replace IKEBindAddress from `172.16.2.100` to `192.168.56.101`, namely from:
+```
+  IKEBindAddress: 172.16.2.100 # Nwu interface  IP address (IKE) on this N3IWF
+```
+into:
+```
+  IKEBindAddress: 192.168.56.101 # Nwu interface  IP address (IKE) on this N3IWF
+```
+## 5. Setting N3IWUE
+
+In n3iwue VM, we need to edit one file:
+
+- `~/n3iwue/config/n3ue.yaml`
```

Comment:
In N3IWUE VM, we need to edit the UE config file `~/n3iwue/config/n3ue.yaml`:

Reply:
To let N3IWUE knows where is the N3IWF is, we need to edit the UE config file `~/n3iwue/config/n3ue.yaml` in N3IWUE VM: